app/controllers/user_sessions_controller.rb

```diff
@@ -50,7 +50,9 @@ def handle_social_login_flow(auth)
         redirect_to return_to + hash_params, notice: "Already linked to another account!"
       end
     else # not signed in
-      @user = User.where(email: auth["info"]["email"]) ? User.find_by(email: auth["info"]["email"]) : @identity.user
+      # if user has the same email for his oauth login and his publiclab website, we can get him using
```

Comment:
We can't write informal text in comments. I have given you text. Kindly paste it directly. Thanks